numpy/core/src/multiarray/datetime.c

```diff
@@ -2347,6 +2347,76 @@ get_tzoffset_from_pytzinfo(PyObject *timezone_obj, npy_datetimestruct *dts)
                  get_datetimestruct_minutes(dts));
 }
 
+
+/*
+ * Gets a tzname str by calling tzname() function on
+ * the Python datetime.tzinfo object.
+ */
+NPY_NO_EXPORT int
+get_tzname_from_pytzinfo(PyObject *timezone_obj, npy_datetimestruct *dts,
+                         char *outstr)
```

Comment:
I think it would be better for this function to just return the `PyObject *`, and let the caller do the memory copy